libs/langchain/langchain/utilities/google_scholar.py

```diff
@@ -53,72 +55,47 @@ class Config:
     @root_validator()
     def validate_environment(cls, values: Dict) -> Dict:
         """Validate that api key and python package exists in environment."""
-        serp_api_key = get_from_dict_or_env(values, "serp_api_key", "SERP_API_KEY")
-        values["SERP_API_KEY"] = serp_api_key
+        values["serp_api_key"] = convert_to_secret_str(
+            get_from_dict_or_env(values, "serp_api_key", "SERPAPI_API_KEY")
+        )
 
         try:
-            from serpapi import GoogleScholarSearch
+            from serpapi import SerpApiClient
 
         except ImportError:
             raise ImportError(
                 "google-search-results is not installed. "
                 "Please install it with `pip install google-search-results"
                 ">=2.4.2`"
             )
-        GoogleScholarSearch.SERP_API_KEY = serp_api_key
-        values["google_scholar_engine"] = GoogleScholarSearch
+
+        values["search_engine"] = SerpApiClient
 
         return values
 
     def run(self, query: str) -> str:
-        """Run query through GoogleSearchScholar and parse result"""
+        """Run query through SerpApi.com and parse result"""
         total_results = []
-        page = 0
-        while page < max((self.top_k_results - 20), 1):
-            # We are getting 20 results from every page
-            # which is the max in order to reduce the number of API CALLS.
-            # 0 is the first page of results, 20 is the 2nd page of results,
-            # 40 is the 3rd page of results, etc.
-            results = (
-                self.google_scholar_engine(  # type: ignore
-                    {
-                        "q": query,
-                        "start": page,
-                        "hl": self.hl,
-                        "num": min(
-                            self.top_k_results, 20
-                        ),  # if top_k_result is less than 20.
-                        "lr": self.lr,
-                    }
-                )
-                .get_dict()
-                .get("organic_results", [])
-            )
-            total_results.extend(results)
-            if not results:  # No need to search for more pages if current page
-                # has returned no results
-                break
-            page += 20
-        if (
-            self.top_k_results % 20 != 0 and page > 20 and total_results
-        ):  # From the last page we would only need top_k_results%20 results
-            # if k is not divisible by 20.
-            results = (
-                self.google_scholar_engine(  # type: ignore
-                    {
-                        "q": query,
-                        "start": page,
-                        "num": self.top_k_results % 20,
-                        "hl": self.hl,
-                        "lr": self.lr,
-                    }
-                )
-                .get_dict()
-                .get("organic_results", [])
-            )
```

Comment:
I replaced this with using the [`.pagination()`](https://github.com/serpapi/google-search-results-python#pagination-using-iterator) method.